pkg/infra/httpclient/max_bytes_reader.go

```diff
@@ -0,0 +1,62 @@
+package httpclient
+
+import (
+	"errors"
+	"io"
+)
+
+// Similar implementation to http/net MaxBytesReader
+// https://pkg.go.dev/net/http#MaxBytesReader
+// What's happening differently here, is that the field that
+// is limited is the response and not the request, thus
+// the error handling/message needed to be accurate.
+
+// MaxBytesReader is similar to io.LimitReader but is intended for
+// limiting the size of incoming request bodies. In contrast to
+// io.LimitReader, MaxBytesReader's result is a ReadCloser, returns a
+// non-EOF error for a Read beyond the limit, and closes the
+// underlying reader when its Close method is called.
+//
+// MaxBytesReader prevents clients from accidentally or maliciously
+// sending a large request and wasting server resources.
+func MaxBytesReader(r io.ReadCloser, n int64) io.ReadCloser {
```

Comment:
Maybe some tests of this would be good as well.

Reply:
I didn't on purpose, since I added some tests for the `response_limit_middleware` which reflect the conditions we want to check, but I will add some here for consistency.